strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/authorization/AuthorizationConfigFileManager.java

```diff
@@ -45,4 +54,54 @@ public ConfigurationResourceResolver getConfigurationResourceResolver()
         return configurationResourceResolver;
     }
 
+    @Override
+    public synchronized void store(AuthorizationConfigDto configuration)
+            throws IOException
+    {
+        Set<SystemRole> changed = getChangedRestrictedRoles(configuration);
+        if (changed.size() > 0)
+        {
+            String changedNames = changed.stream()
+                    .map(SystemRole::toString)
+                    .collect(Collectors.joining(", "));
+            String message = String.format("System roles %s cannot be changed.", changedNames);
+            throw new IOException(message);
+        }
+        super.store(configuration);
+    }
+
+    @VisibleForTesting
+    public Set<SystemRole> getChangedRestrictedRoles(AuthorizationConfigDto configuration)
+            throws IOException
+    {
+        Set<SystemRole> changed = new HashSet<>();
+        Map<String, SystemRole> restricted = SystemRole.getRestricted();
+        for (RoleDto role : configuration.getRoles())
+        {
+            if (!restricted.containsKey(role.getName()))
+            {
+                continue;
+            }
+            SystemRole systemRole = restricted.get(role.getName());
+            // get MD5 hash for the role
+            String hash = null;
+            try {
```

Comment:
Cosmetic: Could you please follow the project's coding convention? Here and across the rest of the code? For examples, braces should be on a new line.

Reply:
God knows I try! I just don't see it :).  Is there any tool to find same line braces?

Reply:
On our [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) there are links to settings files for both Idea and Eclipse. You might want to try those out, although, (admittedly), they are not perfect.